backend/src/akvo/lumen/lib/auth.clj

```diff
@@ -213,16 +217,16 @@
           (handler (assoc request
                           :auth-service (new-auth-service auth-uuid-tree))))))))
 
-(defmethod ig/init-key :akvo.lumen.lib.auth/wrap-auth-datasets  [_ {:keys [tenant-manager flow-api monitoring] :as opts}]
-  (wrap-auth-datasets tenant-manager flow-api (:collector monitoring)))
+(defmethod ig/init-key :akvo.lumen.lib.auth/wrap-auth-datasets  [_ {:keys [tenant-manager authz-service-client monitoring] :as opts}]
+  (wrap-auth-datasets tenant-manager authz-service-client (:collector monitoring)))
 
-(s/def ::flow-api ::c.flow/config)
+(s/def ::authz-service-client any?)
```

Comment:
trying to avoid this `any?` spec I've set up this PR https://github.com/akvo/akvo-lumen/pull/2478

Reply:
Merged, but honestly, I wonder if adding one protocol and one record for that is worth it.

Reply:
yep, I thought the same, that's the reason that I requested your review :) 

anyway we'll have 2 things now, we are sure that our init(iated) system will have an authentication-client-protocol there and we could also mock the component in other envs